apm-agent-plugins/apm-jms-plugin/src/test/java/co/elastic/apm/agent/jms/JmsInstrumentationIT.java

```diff
@@ -134,7 +135,23 @@ public void testQueueSendReceiveOnNonTracedThread() throws Exception {
     @Test
     public void testQueueSendReceiveNoWaitOnNonTracedThread() throws Exception {
         final Queue queue = brokerFacade.createQueue(TEST_Q_NAME);
-        testQueueSendReceiveOnNonTracedThread(() -> brokerFacade.receiveNoWait(queue), queue);
+        testQueueSendReceiveOnNonTracedThread(() -> loopReceive(() -> brokerFacade.receiveNoWait(queue), 3000), queue);
+    }
+
+    // A utility method for testing the receiveNoWait API consistently
+    private Message loopReceive(Callable<Message> receiveMethod, @SuppressWarnings("SameParameterValue") long timeout) throws Exception {
```

Comment:
consider using [Awaitility](https://github.com/awaitility/awaitility/wiki/Usage)

Reply:
Nice! But let's merge this now, and consider next time?